### DISCUSSION TITLE
Where do downloaded .exe files go?

### DISCUSSION THREAD

**Discussion by kneelesh48** [microsoft/winget-cli#1148] [Q&A] [answered]
When I run something like `winget install vlc` it downloads vlc.exe but where is it saved?

I downloaded a large installer file with winget but the installation failed because my computer was locked when the UAC screen appeared and I just got an error in powershell (this should be fixed).

I'd prefer not to download it again but to use the downloaded exe file.

Please tell me where exactly it is saved.

**Comment by JohnMcPMS** [comment, marked as answer]
Currently the file ends up here:
```
%TEMP%\WinGet
```

**Reply by kneelesh48** [reply]
Thank you

**Reply by soroshsabz** [reply]
Thanks

https://superuser.com/a/1686874/167405

**Reply by IkelAtomig** [reply]
How long the package is there in Temp directory in case I want to copy the installer file ?

**Reply by cebu23** [reply]
I see that the file lands in the %TEMP% as indicated in the answer, with each installer landing in a subdirectory with its name and version.  However, it looks like those files only persist until the next install (of any package!) is run; i.e. I manually ran upgrades on a number of packages (`winget upgrade Vendor.Package1`, `winget upgrade Vendor2.Package2`, etc. rather than `winget upgrade --all`), and while all of the subdirectories existed, only the Vendor2.Package2 installer file was still present on the disk.

Could we perhaps get an option on both `upgrade` and `install` that retains the installer indefinitely?  I like to keep the latest installer (and sometimes a back version or two) available on the NAS, both for installing to other systems and as a "just in case" if an installer can no longer be downloaded, the new version is wonky, or (the most likely) I do something stupid.

**Reply by IkelAtomig** [reply]
That's reasonable and would be useful to copy and install in other devices.

**Reply by 9node** [reply]
is there a way to clear winget specific cache and downloaded files ?

**Reply by JohnMcPMS** [reply]
None provided directly, but you may be able to safely delete the directory `%TEMP%\WinGet`.  I say "may" because some installers have been known to just plop their files next to the installer, making that location their home.  Those are inherently fragile and `%TEMP%` is typically emptied by Windows upgrades (OS swap; the big ones that we only do once a year [circa 2024]).

So the short (and mostly harmless) answer is: Delete `%TEMP%\WinGet`

The default location for `winget download` is not the same as the temporary installer location; it can be found with `winget --info`.  That currently shows me:
```
Installer Downloads                %USERPROFILE%\Downloads
```
